src/main/java/com/newrelic/experts/jenkins/extensions/EventCollectorWork.java

```diff
@@ -77,49 +77,59 @@ public long getRecurrencePeriod() {
   
   @Override
   protected void doRun() throws Exception {
-    final String methodName = "doRun";
-    final boolean isLoggingTrace = LOGGER.isLoggable(Level.FINE);
-    final boolean isLoggingDebug = LOGGER.isLoggable(Level.FINEST);
+    synchronized (this.recorder) {
```

Comment:
@dalvizu good catch on the need to synchronize access to the event recorder.  However, this critical section is overly large.  I would prefer to modify the `EventRecorderImpl` by adding a single `synchronized private void recordEvent(Event event)` method that all the other `record*` methods call after they have prepared the event.  This would keep the critical section as small as possible.  It also means we do not need to push the need to synchronize out to all the clients of the recorder.